### PR TITLE
--tofqdns-min-ttl default depends on --tofqdns-enable-poller

### PR DESCRIPTION
**Pull request by raybejjani**
--tofqdns-min-ttl controls the lower bound to cache DNS information.
This is used to ensure that long-lived connections are not interrupted,
and that restarting cilium will restore enough information. A longer TTL
results in more identities allocated. This becomes a problem when DNS
returns many unique IPs, and the DNS poller will do this when pointed to
certain common domains like S3. In these cases a shorter TTL is better,
as it avoids long policy regeneration times because data expires sooner.
The defaults are now different when the poller is enabled (a shorter 1
hour) versus when it is disabled (the default, 1 week). Both values
remain configurable.

I tried to find a way to do this within viper/sp13 but I don't think it's possible to get the value of --tofqdns-enable-poller before options parsing... so it is a little clunkily hard-coded now.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6599)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-docs-please

**Comment by raybejjani** [comment]
test-me-please

File: pkg/option/config.go [hunk 1]
```diff
@@ -836,14 +836,27 @@ func (c *DaemonConfig) Populate() {
...
 
+	// toFQDNs options
+	// When the poller is enabled, the default MinTTL is lowered. This is to
+	// avoid caching large sets of identities generated by a poller (it runs
+	// every 5s). Without the poller, a longer default is better because it
+	// avoids confusion about dropped connections.
+	c.ToFQDNsEnablePoller = viper.GetBool(ToFQDNsEnablePoller)
+	switch {
+	case viper.GetInt(ToFQDNsMinTTL) != 0: // set by user
+		c.ToFQDNsMinTTL = viper.GetInt(ToFQDNsMinTTL)
```
Comment (tgraf) [inline pkg/option/config.go]: nit: result of first GetInt() call could be cached in variable

**Review by tgraf — APPROVED** [review]

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-docs-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**